deps/chakrashim/lib/chakra_debug.js

```diff
@@ -0,0 +1,1304 @@
+    var bpObject = CallHostFunction(chakraDebug.JsDiagSetBreakpoint,
+      this.scriptObject.GetId(), this.line, this.column);
+    if (!bpObject) {
+      return false;
+    }
+    this.v8breakpoint = {};
+    this.v8breakpoint.number = bpObject.breakpointId;
+    this.v8breakpoint.line = bpObject.line;
+    this.v8breakpoint.column = bpObject.column;
+    this.v8breakpoint.groupId = null;
+    this.v8breakpoint.hit_count = 0;
+    this.v8breakpoint.active = true;
+    this.v8breakpoint.condition = null;
+    this.v8breakpoint.ignoreCount = 0;
+    this.v8breakpoint.actual_locations = [];
+    this.v8breakpoint.actual_locations[0] = {};
+    this.v8breakpoint.actual_locations[0].line = bpObject.line;
+    this.v8breakpoint.actual_locations[0].column = bpObject.column;
+    this.v8breakpoint.actual_locations[0].script_id = bpObject.scriptId;
+    this.v8breakpoint.type = this.GetTypeString();
+    this.v8breakpoint.script_id = bpObject.scriptId;
+    this.v8breakpoint.script_name = this.scriptObject.GetScriptName();
+
+    if (this.type == Debug.ScriptBreakPointType.ScriptRegExp) {
+      this.v8breakpoint.script_regexp = this.target;
+    }
+
+    return true;
+  };
+
+  V8Breakpoint.prototype.IncreaseHit = function() {
+    this.v8breakpoint.hit_count++;
+  };
+
+  V8Breakpoint.prototype.Remove = function() {
+    CallHostFunction(chakraDebug.JsDiagRemoveBreakpoint, this.Id());
+  };
+
+  V8Breakpoint.prototype.Id = function() {
+    return this.v8breakpoint.number;
+  };
+
+  V8Breakpoint.prototype.line = function() {
+    return this.bpObject.line;
+  };
+
+  V8Breakpoint.prototype.column = function() {
+    return this.bpObject.column;
+  };
+
+  V8Breakpoint.prototype.GetProtocolObject = function() {
+    return this.v8breakpoint;
+  };
+
+
+  function V8CommandProcessor() {}
+  // Supported commands
+  V8CommandProcessor.prototype.scripts = function(request, response) {
+    var v8Scripts = DebugManager.ScriptsManager.GetScripts();
+
+    if (request.arguments) {
+      var ids;
+      var filter;
+      var includeSource = false;
+
+      if (request.arguments.ids) {
+        ids = request.arguments.ids;
+      }
+
+      if (request.arguments.includeSource) {
+        includeSource = true;
+      }
+
+      if (request.arguments.filter) {
+        filter = request.arguments.filter;
+      }
+
+      var body = [];
+      v8Scripts.forEach(function(element, index, array) {
+        var found = true;
+
+        if (ids != null) {
+          found = false;
+          for (var i = 0; i < ids.length; ++i) {
+            if (ids[i] == element.GetId()) {
+              found = true;
+              break;
+            }
+          }
+        }
+
+        if (filter != null) {
+          found = false;
+          if (filter == element.GetFileName()) {
+            found = true;
+          }
+        }
+
+        if (found == true) {
+          var scriptObj = element.GetProtocolObject();
+          if (includeSource == true) {
```

Comment:
Sorry I didn't understood what you mean here? found is used to track if a particular scriptid is requested and includeSource is tracking if additionally we need to include completed source of the script or not.

---
In reply to: [82246914](https://github.com/nodejs/node-chakracore/pull/126#discussion_r82246914) [](ancestors = 82246914)